sklearn/decomposition/online_lda.py

```diff
@@ -0,0 +1,699 @@
+
+        if cal_sstats:
+            # This step finishes computing the sufficient statistics for the M step
+            suff_stats = np.zeros(self.components_.shape)
+            for sstats in sstats_list:
+                suff_stats += sstats
+            suff_stats *= self.exp_dirichlet_component_
+        else:
+            suff_stats = None
+
+        return (doc_topic_distr, suff_stats)
+
+    def _em_step(self, X, total_samples, batch_update):
+        """
+        EM update for 1 iteration.
+        update `_component` by batch VB or online VB.
+
+        Parameters
+        ----------
+        X : array-like or sparse matrix, shape=(n_samples, n_features)
+            Document word matrix.
+
+        total_samples : integer
+            Total umber of documents. It is only used when
+            batch_update is `False`.
+
+        batch_update : boolean
+            Parameter that controls updating method.
+            `True` for batch learning, `False` for online learning.
+
+        Returns
+        -------
+        doc_topic_distr : array, shape=(n_samples, n_topics)
+            Unnormalized document topic distribution.
+        """
+
+        # E-step
+        _, suff_stats = self._e_step(X, cal_sstats=True, random_init=True)
+
+        # M-step
+        if batch_update:
+            self.components_ = self.topic_word_prior_ + suff_stats
+        else:
+            # online update
+            # In the literature, the weight is `rho`
+            weight = np.power(self.learning_offset + self.n_iter_, -self.learning_decay)
+            doc_ratio = float(total_samples) / X.shape[0]
+            self.components_ *= (1 - weight)
+            self.components_ += (weight *
+                                 (self.topic_word_prior_ + doc_ratio * suff_stats))
+
+        # update `component_` related variables
+        self.dirichlet_component_ = _log_dirichlet_expectation(self.components_)
+        self.exp_dirichlet_component_ = np.exp(self.dirichlet_component_)
+        self.n_iter_ += 1
+        return
+
+    def _check_non_neg_array(self, X, whom):
+        """
+        check & convert X to csr format, and make sure no negative value in X.
+
+        Parameters
+        ----------
+        X :  array-like or sparse matrix
+
+        """
+        X = check_array(X, accept_sparse='csr')
+        check_non_negative(X, whom)
+        return X
+
+    def fit_transform(self, X, y=None):
+        """
+        Learn a model for X and return the transformed data
+
+        Parameters
+        ----------
+        X : array-like or sparse matrix, shape=(n_samples, n_features)
+            Document word matrix.
+
+        Returns
+        -------
+        doc_topic_distr : array, [n_samples, n_topics]
+            Topic distribution for each document.
+        """
+        self._check_params()
+        X = self._check_non_neg_array(X, "LatentDirichletAllocation.fit_transform")
+        return self.fit(X).transform(X)
```

Comment:
I don't see the necessity of defining this method: I don't think that it provides much value compared to the inherited fit_transform method from the TransformerMixin.